veps/sig-network/ns-admin-nichotplug.md

```diff
@@ -0,0 +1,220 @@
+- Keeps the integration with the dynamic-networks-controller
+
+#### Cons
+ - Will not work for non-migratable VMs
+ - Performs a migration for every hotplug / hotunplug
+ - Requires an API change
+
+## API Examples
+
+<!--
+Tangible API examples used for discussion
+-->
+
+## Alternatives
+
+### Using the dynamic-networks-controller
+
+The dynamic-networks-controller is a controller, running as a [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/),
+listening to changes on Pods and invoking the [Multus v4 CNI](https://github.com/k8snetworkplumbingwg/multus-cni).
+
+It enables NIC hotplug / hotunplug operations in place, e.g. without creating another pod.
+A known limitation of the controller above is that it cannot hotplug interfaces that require invocation of
+[Kubernetes device plugins](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/),
+for example: SR-IOV network interfaces.
+
+#### Pros
+
+- Improved user experience
+- No need for migration when attaching/detaching interfaces using bridge binding
+- Works on non-migratable VMs
+- Does not require changes to kubevirt/kubevirt
+
+#### Cons
+
+- The dynamic-networks-controller is an external component that needs to be deployed and managed
+- Not applicable for hotplugging SR-IOV interfaces
+
+### Providing Namespace Admins the required RBAC Rules
+
+Cluster admins could provide Namespace admins the permissions to create VMIM objects.
+The documentation will be updated to reflect the fact that there are additional RBAC rules missing for namespace
+admins to be able to perform NIC hotplug / unplug.
+
+#### Pros
+
+- Does not require code changes
+
+#### Cons
+
+- Losses the benefits of limiting VMIM creation to cluster admins
+- Requires cluster-admins to take action
+
+## Scalability
+
+N/A.
+
+## Update/Rollback Compatibility
+
+Users will have to configure their KubeVirt configuration similarly to CPU and memory hotplug in order to use the automatic
+migration.
+The manual migration will no longer achieve the desired affect.
+Users using the dynamic-networks-controller will have to adjust their KubeVirt configuration.
+
+## Functional Testing Approach
+
+The migration-based hotplug / hotunplug tests will be adjusted to remove the manual VMIM object creation.
+The in-place hotplug / hotunplug tests will be adjusted to assert that a migration did not occur.
+
+## Implementation Phases
+
+1. Introduce new condition.
+2. Adjust network VMI controller logic to set the condition.
+3. Adjust the workload update controller to consider the new condition.
+4. Adjust the e2e tests to work without dynamic-networks-controller.
+5. Adjust the VM controller to invoke the network controller differently.
+6. Adjust the network VM controller.
+7. Introduce new knob to KubeVirt CRD.
+8. Adjust the VMI controller logic to consider the new knob.
+9. Adjust the e2e tests to work with the dynamic-networks-controller.
```

Comment:
The phases need to present a functional operation, even if limited.
Please re-arange these to provide a functional operation.

Reply:
Done.